### PR TITLE
[ticket/8672] Add class for retrieving imagesize without download

### PR DESCRIPTION
**Pull request by marc1706**
getimagesize() always downloads the complete file before checking
the actual image dimensions. This class will be able to do the same
without having to download possibly large files.

Ticket: https://tracker.phpbb.com/browse/PHPBB3-8672

### CODE REVIEW

File: phpBB/phpbb/upload/imagesize.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	/**
+	 * Get dimensions of PNG image
+	 *
+	 * @param string $filename Filename of image
+	 * @return array|bool Array with image dimensions if successful, false if not
+	 */
+	protected function get_png_size($filename)
+	{
+		$data = file_get_contents($filename, null, null, 0, 24);
```
Comment (MGaetan89) [inline phpBB/phpbb/upload/imagesize.php]: Maybe add a comment to explain the 24?

**Comment by marc1706** [comment]
Removed the WIP label. Please review and comment :)

File: phpBB/includes/functions_upload.php [hunk 2]
```diff
@@ -387,28 +387,26 @@ function move_file($destination, $overwrite = false, $skip_image_check = false,
...
-				$this->width = $this->image_info[0];
-				$this->height = $this->image_info[1];
+			// Get imagesize class
+			$imagesize = new \phpbb\upload\imagesize();
 
-				if (!empty($this->image_info['mime']))
-				{
-					$this->mimetype = $this->image_info['mime'];
-				}
+			if (($this->image_info = $imagesize->get_imagesize($this->destination_file, $this->mimetype)) !== false)
```
Comment (bantu) [inline phpBB/includes/functions_upload.php]: Un-inline assignment?

**Comment by marc1706** [comment]
@bantu removed inline assignment

**Comment by Nicofuma** [comment]
looks good

**Pull request closed without merging**